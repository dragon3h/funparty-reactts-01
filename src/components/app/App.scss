.container {
  margin: 2rem auto;
  width: 60vw;
}