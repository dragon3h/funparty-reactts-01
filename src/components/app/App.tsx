import React from 'react';
import {Route, Switch} from 'react-router-dom';
import {connect} from 'react-redux';

import './App.scss';
import Login from '../login/login.component';
import Home from '../home/home.component';
import { UserLogin } from '../../redux/user/types';

type AppProps = {
  currentUser: UserLogin
};

type AppState = {
  isAuth: boolean
};

class App extends React.Component<AppProps, AppState> {
  
  render() {
    let app;
    if (this.props.currentUser) {
      app = (
          <Home/>
      );
    } else {
      app = (
          <Login/>
      );
    }
    return (
        <div className="container">
          {app}
        </div>
    );
  }
}

const mapStateToProps = (state: any) => {
  return {
    currentUser: state.user
  };
};

export default connect(mapStateToProps)(App);
