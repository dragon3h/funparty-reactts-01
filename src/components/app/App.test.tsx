import React from 'react';
import * as renderer from 'react-test-renderer';

import Root from '../../root';

import App from './App';

test('renders App component without crashes', () => {
  const mainScreen = renderer.create(
      <Root>
        <App />
      </Root>).toJSON();

  expect(mainScreen).toMatchSnapshot();
});

// it('renders without crashing', () => {
//   const tree = ren
// });
