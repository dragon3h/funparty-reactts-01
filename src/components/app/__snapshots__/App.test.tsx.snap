// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component without crashes 1`] = `
<div
  className="container"
>
  <div
    className="login"
  >
    <h1>
      Login
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="username"
          className="form-input"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
        <label
          className=" form-input__label"
        >
          Email
        </label>
      </div>
      <div
        className="input-group"
      >
        <input
          autoComplete="current-password"
          className="form-input"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
        <label
          className=" form-input__label"
        >
          Password
        </label>
      </div>
      <button
        className="custom-button"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
